modules/boot.nix

```diff
@@ -0,0 +1,14 @@
+{...}: {
+  boot = {
+    kernelModules = ["coretemp"];
+    loader = {
+      systemd-boot.enable = true;
+      efi.canTouchEfiVariables = true;
+    };
+    supportedFilesystems = ["zfs"];
+    zfs = {
```

Comment:
Can we split out a separate ZFS module that includes this and any other ZFS-related config?